### PR TITLE
Fixed your issues

### PR DESCRIPTION
**Pull request by Mbucari**
#574 was particularly difficult to nail down.

### CODE REVIEW

**Comment by Mbucari** [comment]
I changed the original title because "Fixing your bugs" had an accusatory tone, but "fixing your issues" sounds even more judgemental 😂

File: Source/LibationWinForms/GridView/GridEntryBindingList.cs [hunk 1]
```diff
@@ -24,7 +24,11 @@ namespace LibationWinForms.GridView
 	internal class GridEntryBindingList : BindingList<IGridEntry>, IBindingListView
 	{
 		public GridEntryBindingList() : base(new List<IGridEntry>()) { }
-		public GridEntryBindingList(IEnumerable<IGridEntry> enumeration) : base(new List<IGridEntry>(enumeration)) { }
+		public GridEntryBindingList(IEnumerable<IGridEntry> enumeration) : base(new List<IGridEntry>(enumeration))
+		{
+			SearchEngineCommands.SearchEngineUpdated += (_,_) => ApplyFilter(FilterString);
```
Comment (rmcrackan) [inline Source/LibationWinForms/GridView/GridEntryBindingList.cs:29]: This isn't functionally wrong, but it feels philosophically wrong. Something will update the object then persist to the database, possibly within the same method. The persistence will say when it's updated and will then update the search engine. It seems like the persistence-update would be the thing to subscribe to. I'm going off of memory though and you've obviously been more boots on the ground lately. Thoughts?

File: Source/LibationUiBase/GridView/GridEntry[TStatus].cs [hunk 2]
```diff
@@ -154,6 +154,10 @@ private void UserDefinedItem_ItemChanged(object sender, string itemName)
 			if (udi.Book.AudibleProductId != Book.AudibleProductId)
 				return;
 
+			//If UserDefinedItem was changed on a different Book instance (such as when batch liberating via menus),
+			//EntryStatu's Book instance will not have the current DB state.
+			Liberate.Book = udi.Book;
```
Comment (rmcrackan) [inline Source/LibationUiBase/GridView/GridEntry[TStatus].cs:159]: That's wacky. I bet tracking down this idiosyncrasy was a challenge and a half.

btw -- typo on `EntryStatu's`

**Comment by rmcrackan** [comment]
> fixing your issues

You don't begin to have the time to fix my issues

Thank you for being an advocate for: do it if and only if it's fun. I've felt guilty lately at how much you've done while I've very much not been in the mood. Especially on this batch when you're fixing the bugs I which submitted instead of fixing myself.

File: Source/LibationUiBase/GridView/GridEntry[TStatus].cs (hunk 2 above)
Comment (rmcrackan) [reply Source/LibationUiBase/GridView/GridEntry[TStatus].cs:159]: Thank you for leaving responsible comments. This is perfect -- it's a note to us later about why this unintuitive thing was done and what exactly was wrong without it.

File: Source/LibationWinForms/GridView/GridEntryBindingList.cs (hunk 1 above)
Comment (Mbucari) [reply Source/LibationWinForms/GridView/GridEntryBindingList.cs:29]: I'm not fond of this solution either, but this bug is a problem...

Here's where we detect changes and re-filter.
https://github.com/rmcrackan/Libation/blob/e52772826af681b3805e292b3bea19602975d305/Source/LibationWinForms/GridView/GridEntryBindingList.cs#L218-L227

The problem is that the item changes before the search has been re-indexed. I feel like this has been an ongoing problem.

Anyway, I'll download an old release tomorrow to see if I can find a time when it was working properly and see what we did then. I'm not in the best state to be thinking about this now.

**Comment by Mbucari** [comment]
> You don't begin to have the time to fix my issues

Don't feel bad about that. It's people like us who keep the mental health profession thriving. We're job creators!

File: Source/LibationUiBase/GridView/GridEntry[TStatus].cs (hunk 2 above)
Comment (Mbucari) [reply Source/LibationUiBase/GridView/GridEntry[TStatus].cs:159]: Yeah, I looked at that and thought "I'll probably try to remove this later". Lol

File: Source/LibationWinForms/GridView/GridEntryBindingList.cs (hunk 1 above)
Comment (rmcrackan) [reply Source/LibationWinForms/GridView/GridEntryBindingList.cs:29]: Will `BookUserDefinedItemCommitted` accomplish the same?

> I'm not in the best state to be thinking about this now

Take care of you. Nothing about this is urgent.

**Pull request merged**